### ISSUE TITLE
Windows/GTK3 Runtime Error: Procedure entry point deflateSetHeader...

### ISSUE THREAD

**Issue by fralau** [Kozea/WeasyPrint#1551]
First of all, let me thank for the great library, which I have run without problems on MacOS and Linux.

Windows requires GTK3 Runtime, which worked in one case.

On one specific Windows 10 platform (weasyprint-54.0, python 3.9), I get the however the following error:
<img width="287" alt="Error" src="https://user-images.githubusercontent.com/36767124/151200692-94bf8d97-1389-4ef5-be51-8a6a4b8e3e8a.png">

And finally, 

```
 File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\timesheets-0.2.3-py3.9.egg\timesheets\produce_reports.py", line 20, in <module>
    import weasyprint  # to convert to pdf
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\weasyprint-54.0-py3.9.egg\weasyprint\__init__.py", line 325, in <module>
    from .css import preprocess_stylesheet  # noqa isort:skip
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\weasyprint-54.0-py3.9.egg\weasyprint\css\__init__.py", line 27, in <module>
    from . import computed_values, counters, media_queries
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\weasyprint-54.0-py3.9.egg\weasyprint\css\computed_values.py", line 16, in <module>
    from ..text.ffi import ffi, pango, units_to_double
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\weasyprint-54.0-py3.9.egg\weasyprint\text\ffi.py", line 406, in <module>
    pango = _dlopen(
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\weasyprint-54.0-py3.9.egg\weasyprint\text\ffi.py", line 390, in _dlopen
    return ffi.dlopen(names[0])  # pragma: no cover
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\cffi-1.15.0-py3.9-win-amd64.egg\cffi\api.py", line 150, in dlopen
    lib, function_cache = _make_ffi_library(self, name, flags)
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\cffi-1.15.0-py3.9-win-amd64.egg\cffi\api.py", line 832, in _make_ffi_library
    backendlib = _load_backend_lib(backend, libname, flags)
  File "C:\Users\xxxxx\scoop\apps\python\current\lib\site-packages\cffi-1.15.0-py3.9-win-amd64.egg\cffi\api.py", line 827, in _load_backend_lib
    raise OSError(msg)
OSError: cannot load library 'pango-1.0-0': error 0x7e.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'pango-1.0-0'
```

I realize this might be a problem in any random place. Any idea of where I should start?

**Comment by fralau** [comment]
For reference, could that be an issue connected to the **zlib** compression utility[(see website)](https://zlib.net/) as suggested by [a 2017 issue on the Gnome mail list?](https://mail.gnome.org/archives/gtk-app-devel-list/2017-January/msg00018.html)

And if yes, how could one go about solving that issue?

**Comment by liZe** [comment]
Hi @fralau,

Yes, it’s probably a problem related to the zlib library. There’s probably another version of the zlib library installed on your system, and it’s used instead of the one proposed by GTK.

On Windows, we’ve given Python a *hint* about where to find libraries: see b7fdfd7. Unfortunately, it looks like it’s not enough.

The "solution" is to find where the other zlib library is installed, and see why it’s used instead of GTK’s one.

**Comment by fralau** [comment]
Hi @liZe 

> The "solution" is to find where the other zlib library is installed, and see why it’s used instead of GTK’s one.

That seems to make sense. But were do I start from?

**Comment by liZe** [comment]
> That seems to make sense. But were do I start from?

It’s hard to say, because I don’t use Windows :/.

> Is that a Windows action, a python action?

I suppose that the other zlib dll is installed in a folder that’s listed in the `PATH` environment variable. You can [print `PATH`](https://www.shellhacks.com/windows-list-environment-variables-cmd-powershell/) and search in the listed folders.

**Comment by liZe** [comment]
@fralau Did you find a solution for your problem?

**Comment by fralau** [comment]
Not yet. We will do an on-site investigation (since it is not easily reproducible) and see if we can come up with a reasonable way to solve that library conflict.

**Comment by mbutigan** [comment]
Quick and easy (also dirty) way to solve this problem is to copy zlib1.dll from GTK3 Runtime bin folder into the folder with exe file you are trying to run.

Root of the problem is that some Windows program also uses zlib1.dll name for a completely different purpose, and it prevents using of zlib1.dll from GTK3 installation.

**Comment by fralau** [comment]
Thanks a lot! This was a tricky situation on a target Windows machine with not much possibility for support and investigation. Basically, either the install of a piece of software works reliably on deployment machines, or it is toast. 

We had to go with wkhtmltopdf and python/pdfkit because it has its own reliable installer.  

This is a little sad, because I prefered Weasyprint 😞

**Comment by liZe** [comment]
> This is a little sad, because I prefered Weasyprint disappointed

Thanks for this information. Let’s hope that we’ll be able to fix #1550 soon.
